test/test_torch.py

```diff
@@ -17532,6 +17532,19 @@ def test_pickle_gradscaler(self, device):
                 if lazy_init_scale:
                     self.assertEqual(b.scale(torch.tensor([4.0], dtype=torch.float32, device=device)), 12.0)
 
+    @dtypesIfCUDA(torch.half, torch.float, torch.double,
+                  torch.int8, torch.short, torch.int, torch.long)
+    @dtypes(torch.float, torch.double,
+            torch.int8, torch.short, torch.int, torch.long)
+    def test_nansum(self, device, dtype):
+        x = (torch.randn(3, 3))
+        if dtype in [torch.half, torch.float, torch.double]:
+            x[x < 0.2] = float('nan')
+        # Randomly scale the values
+        x = (x * random.randint(10, 100)).tolist()
+
+        self.compare_with_numpy(torch.nansum, np.nansum, x, device, dtype)
```

Comment:
Like with count_nonzero, we should be sure to test a variety of dim arguments. This function is even more complicated, however, and should also test the dtype and keepdim kwargs. keepdim is easy to test because it's just true/false. dtype is a little trickier. In particular, what happens if you specify a float dtype when given an integer tensor? That should just dispatch to sum, anyway, right? But it'd be good to verify. Some other interesting dtype cases would be if you specify a dtype with less precision than the input (float input, dtype=half) and if you have a very low precision input with a high precision dtype (input=half, dtype=double). If you support complex dtypes, then what happens if you have a complex input but a float dtype, or a float input with a complex dtype? Etc.